x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java

```diff
@@ -196,7 +248,13 @@ public void validate(final ByteSizeValue value, final Map<Setting<?>, Object> se
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public FrozenCacheService(NodeEnvironment environment, Settings settings, ThreadPool threadPool) {
         this.currentTimeSupplier = threadPool::relativeTimeInMillis;
-        this.cacheSize = SNAPSHOT_CACHE_SIZE_SETTING.get(settings).getBytes();
+        long totalFsSize;
+        try {
+            totalFsSize = FsProbe.getTotal(Environment.getFileStore(environment.nodeDataPaths()[0]));
+        } catch (IOException e) {
+            throw new IllegalStateException("unable to probe size of filesystem [" + environment.nodePaths()[0] + "]");
```

Comment:
also change to environment.nodeDataPaths

Reply:
Thanks, sorry for missing that one 1b3e031550a7f57bc1ac69a3803867bddc2c54c3